server/src/main/java/org/elasticsearch/action/admin/cluster/desiredbalance/DesiredBalanceRequest.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.action.admin.cluster.desiredbalance;
```

Comment:
The rest of this code is just in the allocation package, we're not (currently) giving it its own package, so I suggest doing the same here (applies to all the classes in this new package)

```suggestion
package org.elasticsearch.action.admin.cluster.allocation;
```